src/frontend/test_runner/tests/testdata/join.yaml

```diff
@@ -275,3 +275,33 @@
     create table b(b1 int);
     select * from a join lateral (select * from b where a1 = b1);
   binder_error: 'Feature is not yet implemented: lateral subqueries are not yet supported, Tracking issue: https://github.com/singularity-data/risingwave/issues/3815'
+- sql: |
+    create table t1(x int, y int);
+    create table t2(x int, y int);
+    select * from t1, t2 where t1.x + t1.y = t2.x + t2.y;
+  optimized_logical_plan: |
+    LogicalJoin { type: Inner, on: ((t1.x + t1.y) = (t2.x + t2.y)) }
+      LogicalProject { exprs: [t1.x, t1.y, (t1.x + t1.y)] }
+        LogicalScan { table: t1, columns: [x, y] }
+      LogicalProject { exprs: [t2.x, t2.y, (t2.x + t2.y)] }
+        LogicalScan { table: t2, columns: [x, y] }
+- sql: |
+    create table t1(x int, y int);
+    create table t2(x float, y int);
+    select * from t1, t2 where t1.x + t1.y = t2.x;
+  optimized_logical_plan: |
+    LogicalJoin { type: Inner, on: ((t1.x + t1.y)::Float64 = t2.x) }
+      LogicalProject { exprs: [t1.x, t1.y, (t1.x + t1.y)::Float64] }
+        LogicalScan { table: t1, columns: [x, y] }
+      LogicalProject { exprs: [t2.x, t2.y, t2.x] }
```

Comment:
This is an unnecessary project created by the bug pointed out above.

Reply:
Actually, this is an unnecessary project but not a bug. we just append left_exprs and right_exprs to left project and right project respectively, so the condition of join refers to them. We can remove the right project in this case, but we just keep it for simplicity now.

Reply:
Ok, so we will depend on some other rule (e.g. project join merge, project scan merge) to remove the project afterwards?

Reply:
If we want to remove this project, I think we should change the indices of LogicalJoin's `on`.

Reply:
Why can't we just make things simpler and don't introduce unnecessary project? It doesn't seem like we need to always symmetrically change LHS and RHS. The LHS can have no exprs etc.

Reply:
> Why can't we just make things simpler and don't introduce unnecessary project

We can definitely make the plan simpler, at the cost of making the code logic more complicated. It is still worth it, but not necessarily in the first PR that enables this feature. It is easier to review if we do less in each PR.